File: backend/internal/core/ports/analytics.go
package ports

import (
	"context"

	"github.com/salonflow/salonflow-track/internal/core/domain"
)

// AnalyticsRepository provides read-only analytics queries across all domain tables.
type AnalyticsRepository interface {
	// Dashboard
	GetDashboardStats(ctx context.Context) (*domain.DashboardStats, error)
	GetKPIMetrics(ctx context.Context, dateFrom, dateTo string) (*domain.KPIMetrics, error)

	// Revenue
	GetRevenueReport(ctx context.Context, dateFrom, dateTo string, groupBy string) (*domain.RevenueReport, error)

	// Customer
	GetCustomerReport(ctx context.Context, dateFrom, dateTo string) (*domain.CustomerReport, error)

	// Staff
	GetStaffReport(ctx context.Context, dateFrom, dateTo string) (*domain.StaffReport, error)

	// Service
	GetServiceReport(ctx context.Context, dateFrom, dateTo string) (*domain.ServiceReport, error)

	// Expense
	GetExpenseReport(ctx context.Context, dateFrom, dateTo string) (*domain.ExpenseReport, error)

	// Inventory
	GetInventoryReport(ctx context.Context) (*domain.InventoryReport, error)

	// P&L
	GetProfitLossReport(ctx context.Context, dateFrom, dateTo string, groupBy string) (*domain.ProfitLossReport, error)
}

File: frontend/src/services/analytics.ts
import type { DashboardStats, KPIMetrics, RevenueReport, CustomerReport, StaffAnalyticsReport, ServiceReport, ExpenseAnalyticsReport, InventoryAnalyticsReport, ProfitLossReport } from '@/types'

export interface DateRangeParams {
  date_from?: string
  date_to?: string
  group_by?: string
}

export async function getDashboardStats(): Promise<DashboardStats> {
  return window.go.main.AnalyticsService.GetDashboard()
}

export async function getKPIMetrics(params: DateRangeParams = {}): Promise<KPIMetrics> {
  return window.go.main.AnalyticsService.GetKPIs(params.date_from || '', params.date_to || '')
}

export async function getRevenueReport(params: DateRangeParams = {}): Promise<RevenueReport> {
  return window.go.main.AnalyticsService.GetRevenueReport(params.date_from || '', params.date_to || '', params.group_by || 'day')
}

export async function getCustomerReport(params: DateRangeParams = {}): Promise<CustomerReport> {
  return window.go.main.AnalyticsService.GetCustomerReport(params.date_from || '', params.date_to || '')
}

export async function getStaffReport(params: DateRangeParams = {}): Promise<StaffAnalyticsReport> {
  return window.go.main.AnalyticsService.GetStaffReport(params.date_from || '', params.date_to || '')
}

export async function getServiceReport(params: DateRangeParams = {}): Promise<ServiceReport> {
  return window.go.main.AnalyticsService.GetServiceReport(params.date_from || '', params.date_to || '')
}

export async function getExpenseReport(params: DateRangeParams = {}): Promise<ExpenseAnalyticsReport> {
  return window.go.main.AnalyticsService.GetExpenseAnalytics(params.date_from || '', params.date_to || '')
}

export async function getInventoryReport(): Promise<InventoryAnalyticsReport> {
  return window.go.main.AnalyticsService.GetInventoryReport()
}

export async function getProfitLossReport(params: DateRangeParams = {}): Promise<ProfitLossReport> {
  return window.go.main.AnalyticsService.GetProfitLossReport(params.date_from || '', params.date_to || '', params.group_by || 'month')
}
